src/analysis/processing/qgsalgorithmroundness.cpp

```diff
@@ -0,0 +1,103 @@
+/***************************************************************************
+                         qgsalgorithmroundness.cpp
+                         ---------------------
+    begin                : Septembre 2021
+    copyright            : (C) 2021 by Antoine Facchini
+    email                : antoine dot facchini @oslandia dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmroundness.h"
+#include "qgscurvepolygon.h"
+
+///@cond PRIVATE
+
+QString QgsRoundnessAlgorithm::name() const
+{
+  return QStringLiteral( "roundness" );
+}
+
+QString QgsRoundnessAlgorithm::displayName() const
+{
+  return QObject::tr( "Roundess" );
+}
+
+QStringList QgsRoundnessAlgorithm::tags() const
+{
+  return QObject::tr( "roundess" ).split( ',' );
+}
+
+QString QgsRoundnessAlgorithm::group() const
+{
+  return QObject::tr( "Vector geometry" );
+}
+
+QString QgsRoundnessAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeometry" );
+}
+
+QString QgsRoundnessAlgorithm::outputName() const
+{
+  return QObject::tr( "Roundess" );
+}
+
+QString QgsRoundnessAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Adds in the attribute table the roundness of each feature. The input vector layer must contain polygons." );
+}
+
+QString QgsRoundnessAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Calculates the roundness of each feature in a polygon vector layer." );
```

Comment:
```suggestion
  return QObject::tr( "Calculates how close each feature shape is to a circle in a polygon vector layer." );
```
Let's explain what it's about.